exercicios/para-casa/sistema_de_alunas.SamanthaTeng.py

```diff
@@ -0,0 +1,176 @@
+from dataset_alunas import dataset
+dataset = {} #Dicionario com escopo global
+def main():
+    print("\n---  Seja bem vinda a Escola do Reprograma!  ---")
+    print("Sistema de informações de alunas")
+    
+    while True:
+        cod_opcao = obter_opcao()
+        
+        if cod_opcao == 1: incluir_nova_aluna()
+        elif cod_opcao == 2: consultar_lista_alunas()
+        elif cod_opcao == 3: consultar_faltas_aluna()
+        elif cod_opcao == 4: consultar_notas_aluna()
+        elif cod_opcao == 5: consultar_status_aprovacao()
+        elif cod_opcao == 6: print("Encerrando o programa..."); break
+
+def obter_opcao():
+    codigo_opcao = 0
+
+    while codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+        try:
+            codigo_opcao = int(input("\nEscolha uma opção:\n"
+                                    "1 - Incluir uma nova aluna\n"
+                                    "2 - Consultar lista de alunas\n"
+                                    "3 - Consultar faltas da aluna\n"
+                                    "4 - Consultar notas da aluna\n"
+                                    "5 - Consultar status de aprovação\n"
+                                    "6 - Sair do sistema\n"
+                                    "Opção: "))
+                
+            if codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+                print("Opção inválida. Por favor, escolha uma opção válida (1 a 5).\n")
+        except ValueError:
+            print("Entrada inválida. Por favor, digite um número inteiro.\n")
+            
+        return codigo_opcao
+
+def incluir_nova_aluna():
+    print("Insira os seguintes dados: ")
+    nome = input("Nome da aluna: ") 
+    sobrenome = input ("Sobrenome: ")
+    turma = input("Turma da aluna: ") 
+    lista_presenca = obter_presenca() 
+    nota_participacao = float(input("Participação da aluna: "))
+    notas = obter_notas() 
+    salvar_dados_aluna(nome,sobrenome, turma, notas, lista_presenca, nota_participacao) 
+    return nome
+
+
+def obter_presenca():
+    quantidade_aulas = input("Quantidade de aulas: ") 
+    aulas = [] 
+    
+    for contador in range(int(quantidade_aulas)): 
+        
+        while True:    
+            entrada = input(f"Insira a presença da aula #{contador + 1}: ") 
+            try: 
+                presenca = eval(entrada) 
+                aulas.append(presenca) 
+                break 
+            except NameError: 
+                print("Entrada inválida. Por favor, insira True ou False.")
```

Comment:
Para evitar erros e simplificar o código, deixo uma nova sugestão:

```suggestion
      while True:
          entrada = input(f"Insira a presença da aula #{contador + 1} (True/False): ").strip().lower()
          if entrada == 'true':
              aulas.append(True)
              break
          elif entrada == 'false':
              aulas.append(False)
              break
          else:
              print("Entrada inválida. Por favor, insira True ou False.")
```